operators/pkg/controller/elasticsearch/certificates/transport/pod_secret.go

```diff
@@ -5,81 +5,76 @@
-
-	// LabelTransportCertificateType is a label key indicating what the transport-certificates secret is used for
-	LabelTransportCertificateType = "transport.certificates.elasticsearch.k8s.elastic.co/certificate-type"
-	// LabelTransportCertificateTypeElasticsearchAll is the LabelTransportCertificateType value used for Elasticsearch
-	LabelTransportCertificateTypeElasticsearchAll = "elasticsearch.all"
+	// LabelCertificateType is a label key that specifies what type of certificates the secret contains
+	LabelCertificateType = "certificates.elasticsearch.k8s.elastic.co/type"
+	// LabelCertificateTypeTransport is the LabelCertificateType value used for transport certificates
+	LabelCertificateTypeTransport = "transport"
 
 	// LastCSRUpdateAnnotation is an annotation key to indicate the last time this secret's CSR was updated
 	LastCSRUpdateAnnotation = "transport.certificates.elasticsearch.k8s.elastic.co/last-csr-update"
 )
 
-// EnsureTransportCertificateSecretExists ensures the existence of the corev1.Secret that at a later point in time will
-// contain the transport certificates.
+// EnsureTransportCertificateSecretExists ensures the existence and Labels of the corev1.Secret that at a later point
+// in time will contain the transport certificates.
 func EnsureTransportCertificateSecretExists(
 	c k8s.Client,
 	scheme *runtime.Scheme,
-	owner metav1.Object,
+	es v1alpha1.Elasticsearch,
 	pod corev1.Pod,
-	certificateType string,
-	labels map[string]string,
 ) (*corev1.Secret, error) {
-	secretRef := types.NamespacedName{
-		Namespace: pod.Namespace,
-		Name:      name.TransportCertsSecret(pod.Name),
-	}
-
-	var secret corev1.Secret
-	if err := c.Get(secretRef, &secret); err != nil && !apierrors.IsNotFound(err) {
-		return nil, err
-	} else if apierrors.IsNotFound(err) {
-		secret = corev1.Secret{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      secretRef.Name,
-				Namespace: secretRef.Namespace,
+	expected := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: pod.Namespace,
+			Name:      name.TransportCertsSecret(pod.Name),
 
-				Labels: map[string]string{
-					// store the pod that this Secret will be mounted to so we can traverse from secret -> pod
-					LabelAssociatedPod:            pod.Name,
-					LabelSecretUsage:              LabelSecretUsageTransportCertificates,
-					LabelTransportCertificateType: certificateType,
-				},
+			Labels: map[string]string{
+				// a label that allows us to list secrets of this type
+				LabelCertificateType: LabelCertificateTypeTransport,
+				// a label that references the pod so we can look up the pod from this secret
+				label.PodNameLabelName: pod.Name,
+				// a label to tells us which cluster this pod is part of
+				label.ClusterNameLabelName: es.Name,
 			},
-		}
-
-		// apply any provided labels
-		for key, value := range labels {
-			secret.Labels[key] = value
-		}
-
-		if err := controllerutil.SetControllerReference(owner, &secret, scheme); err != nil {
```

Comment:
are we not setting the controller reference anymore? Sounds useful to garbage collect these secrets automatically.

Reply:
We're setting the owner through the `reconciler.ReconcileResource` parameter `Owner`. See https://github.com/elastic/cloud-on-k8s/pull/999/files#diff-f8dc8a2276eef8099852b2a7171621bfR59

Reply:
oh yes missed that!